modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/PageMemorySortedIndexStorage.java

```diff
@@ -298,76 +276,74 @@ public void close() {
 
         @Override
         public boolean hasNext() {
-            if (!closeBusyLock.enterBusy()) {
-                throwStorageClosedException();
-            }
-
-            try {
-                advanceIfNeeded();
+            return busy(() -> {
```

Comment:
What's the point of all these changes?

Reply:
Same, but why not use a generic template for this class?